fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -563,9 +565,26 @@ RecordCoreException newPartlyBuildException(boolean continuedBuild,
                     .toString());
         }
 
+        validateTimeLimit();
```

Comment:
I guess there's some more complicated version of this where the function factors in the fact that this build is about to wait `toWait` milliseconds, and then it throws a time limit exception before the time has actually elapsed if the extra time would cause it to exceed the time limit. I don't think that it's strictly necessary, and it may be better to leave it simpler, but food for thought.

Reply:
I like your suggestion! 
Implemented.